src/indexer/shared/entities/asset.ts

```diff
@@ -28,14 +30,12 @@ export const insertAssetIfNotExists = async ({
 
   const assetData = await getAssetData(assetAddress, context);
 
-  const poolAddress = assetData.pool.toLowerCase() as `0x${string}`;
-
   const isToken0 =
     assetAddress.toLowerCase() < assetData.numeraire.toLowerCase();
 
   return await db.insert(asset).values({
     ...assetData,
-    poolAddress,
+    poolAddress: poolAddress ?? assetData.pool.toLowerCase() as `0x${string}`,
```

Comment:
@RhizoNymph tl;dr for this change is that in cases where the pool is multicurve, the assetData.pool returns the token address... so when we create the asset we need to make sure we explicitly pass the poolId as the identifier